src/hotspot/share/gc/parallel/psPromotionLAB.inline.hpp

```diff
@@ -33,9 +33,9 @@ HeapWord* PSYoungPromotionLAB::allocate(size_t size) {
   // Can't assert this, when young fills, we keep the LAB around, but flushed.
   // assert(_state != flushed, "Sanity");
   HeapWord* obj = top();
-  HeapWord* new_top = obj + size;
-  // The 'new_top>obj' check is needed to detect overflow of obj+size.
-  if (new_top > obj && new_top <= end()) {
+  // Pointer overflow check is needed here.
+  if (end() >= obj && size <= (size_t)(end() - obj)) {
```

Comment:
Same suggestions here as for the corresponding changes to the .hpp file.

Reply:
Got it. Thanks.